web/src/views/CreateDandisetView/CreateDandisetView.vue

```diff
@@ -117,6 +134,7 @@ export default defineComponent({
   setup(props, ctx) {
     const name = ref('');
     const description = ref('');
+    const license = ref('');
```

Comment:
```suggestion
    const license = ref<License>();
```

This is what is causing the type-check failures. `license` is implicitly defined as type `Ref<string>` here, but the schema dictates that it's an array of string literals, which is encapsulated in the `License` type. 

Making this change will make `license` instead have a type of `Ref<License | undefined>`, which works as long as you also update `saveDisabled` to return true if the dandiset is not embargoed and has an undefined license (which we want to do anyways), i.e. add `|| (!embargoed.value && !license.value)` to it (github code review unfortunately won't let me make that a formal suggestion b/c this PR doesn't modify that code yet).

Reply:
did you have in mind `ref<LicenseType>()`?

Reply:
No, `ref<License>()` (you would also have to add that to the import statement). `LicenseType` refers to a string, while `License` is an array of `LicenseTypes` that is populated by the `v-select` (see `schema.ts` for those type definitions)